middleware/http/wasm/httpwasm.go

```diff
@@ -0,0 +1,162 @@
+	// Path is where to load a `%.wasm` file that implements the guest side of
+	// the handler protocol. No default.
+	Path string `json:"path"`
+
+	// guest is WebAssembly binary implementing the handler guest, loaded from Path.
+	guest []byte
+}
+
+// middleware is a wasm basic middleware.
+type middleware struct {
+	logger logger.Logger
+}
+
+// NewMiddleware returns a new wasm basic middleware.
+func NewMiddleware(logger logger.Logger) dapr.Middleware {
+	return &middleware{logger: logger}
+}
+
+// GetHandler returns the HTTP handler provided by wasm basic middleware.
+func (m *middleware) GetHandler(metadata dapr.Metadata) (func(next http.Handler) http.Handler, error) {
+	rh, err := m.getHandler(metadata)
+	if err != nil {
+		return nil, err
+	}
+	runtime.SetFinalizer(rh, (*requestHandler).Close)
+	return rh.requestHandler, nil
+}
+
+// getHandler is extracted for unit testing.
+func (m *middleware) getHandler(metadata dapr.Metadata) (*requestHandler, error) {
+	meta, err := m.getMetadata(metadata)
+	if err != nil {
+		return nil, fmt.Errorf("wasm basic: failed to parse metadata: %w", err)
+	}
+
+	var stdout, stderr bytes.Buffer
+	mw, err := wasmnethttp.NewMiddleware(ctx, meta.guest,
+		handler.Logger(m),
+		handler.ModuleConfig(wazero.NewModuleConfig().
+			WithStdout(&stdout).  // reset per request
+			WithStderr(&stderr)), // reset per request
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: meta.PoolSize
+
+	return &requestHandler{mw: mw, logger: m.logger, stdout: &stdout, stderr: &stderr}, nil
+}
+
+// IsEnabled implements the same method as documented on api.Logger.
+func (m *middleware) IsEnabled(api.LogLevel) bool {
+	return true // l.logger.GetOutputLevel()
+}
+
+// Log implements the same method as documented on api.Logger.
+func (m *middleware) Log(_ context.Context, level api.LogLevel, message string) {
+	switch level {
+	case api.LogLevelError:
+		m.logger.Error(message)
+	case api.LogLevelWarn:
+		m.logger.Warn(message)
+	case api.LogLevelInfo:
+		m.logger.Info(message)
+	case api.LogLevelDebug:
+		m.logger.Debug(message)
+	default: // same as api.LogLevelNone
+		return
+	}
+}
+
+func (m *middleware) getMetadata(metadata dapr.Metadata) (*middlewareMetadata, error) {
+	b, err := json.Marshal(metadata.Properties)
+	if err != nil {
+		return nil, err
+	}
+
+	var data middlewareMetadata
+	err = json.Unmarshal(b, &data)
+	if err != nil {
+		return nil, err
+	}
+
+	if data.Path == "" {
+		return nil, errors.New("missing path")
+	}
+
+	data.guest, err = os.ReadFile(data.Path)
+	if err != nil {
+		return nil, fmt.Errorf("error reading path: %w", err)
+	}
+
+	return &data, nil
+}
+
+type requestHandler struct {
+	mw             wasmnethttp.Middleware
+	logger         logger.Logger
+	stdout, stderr *bytes.Buffer
+}
+
+func (rh *requestHandler) requestHandler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		h := rh.mw.NewHandler(ctx, next)
```

Comment:
Can't you get the context from `r.Context()`? This would be a context tied to the request (well, in theory.... in practice it isn't while we're [stuck with fasthttp](https://github.com/dapr/dapr/issues/4979), but that's a different story :) )